synapse/http/server.py

```diff
@@ -296,8 +298,14 @@ def _async_render(self, request):
         # here. If it throws an exception, that is handled by the wrapper
         # installed by @request_handler.
 
+        def _unquote(s):
+            if PY3:
+                return urllib.parse.unquote(s)
+            else:
+                return urllib.parse.unquote(s.encode('ascii')).decode('utf8')
```

Comment:
why do we `encode('ascii')` here?

Reply:
URLs are 7-bit ASCII.

Reply:
ugh, what a mess. thanks for clarifying this.